In [7]:
# импорт библиотеки pandas
import pandas as pd

In [8]:
#чтение датасета с помощью библиотеки pandas
df = pd.read_csv('5music.csv')

## Вывести первые 20 строк с помощью head:

In [9]:
df.head(20)

userID                                          trackname  \
0   FFB692EC                                  Kamigata To Boots   
1   55204538                        Delayed Because of Accident   
2     20EC38                                  Funiculì funiculà   
3   A3DD03C9                              Dragons in the Sunset   
4   E2DC1FAE                                        Soul People   
5   842029A1                                          Преданная   
6   4CB90AA5                                               True   
7   F03E1C1F                                   Feeling This Way   
8   8FA1D3BE                           И вновь продолжается бой   
9   E772D5C0                                          Pessimist   
10  BC5A3A29                                       Gool la Mita   
11  8B5192C0  Is There Anybody Out There? (Panoramic Paralys...   
12  FF3FD2BD                                Мина син генэ кирэк   
13  CC782B0F                               After School Special   
14  94EB25C2                         Make Love Whenever You Can   
15  E3C5756F                                                NaN   
16  81D05C7D                                     SLAVES OF FEAR   
17  39DE290E                                           Веснянка   
18  58AE138A                                         Pat-a-Cake   
19  772F5B59                                          Sweetback   

              Artistname        genre            City        time    Dayweek;;  
0       The Mass Missile         rock  Saint-Petersburg  20:28:33  Wednesday;;  
1       Andreas Rönnberg         rock            Moscow  14:07:09     Friday;;  
2            Mario Lanza          pop  Saint-Petersburg  20:58:07  Wednesday;;  
3             Fire + Ice         folk  Saint-Petersburg  08:37:09     Monday;;  
4             Space Echo        dance            Moscow  08:34:34     Monday;;  
5              IMPERVTOR       rusrap  Saint-Petersburg  13:09:41     Friday;;  
6           Roman Messer        dance            Moscow  13:00:07  Wednesday;;  
7        Polina Griffith        dance            Moscow  20:47:49  Wednesday;;  
8                    NaN       ruspop            Moscow  09:17:40     Friday;;  
9                    NaN        dance  Saint-Petersburg  21:20:49  Wednesday;;  
10   Shireen Abdul Wahab        world            Moscow  14:08:42     Monday;;  
11  Pink Floyd Floydhead   electronic            Moscow  13:47:49     Monday;;  
12        Ильдар Хакимов          pop            Moscow  09:19:49     Monday;;  
13  Detroit Grand Pubahs        dance            Moscow  20:04:12     Friday;;  
14             Arabesque          pop  Saint-Petersburg  13:22:08  Wednesday;;  
15                   NaN          NaN            Moscow  09:24:51     Monday;;  
16                HEALTH  alternative  Saint-Petersburg  20:54:48     Monday;;  
17         Веданъ Колодъ         folk  Saint-Petersburg  14:36:47     Monday;;  
18     Mother Goose Club     children  Saint-Petersburg  20:04:56     Monday;;  
19    The Fabulous Three          rnb            Moscow  13:56:42  Wednesday;;

## Выполнить обзор данных – опишите колонки и вашу предметную область: 

### Предметаная обласать - музыка. 
Мы можем узнать название песни, исполнителя, жанр, идентификатор пользователя. 

Насчет времени сложно сказать, что это продолжительность песни, возможно, это время проигрывания, например на радио, время, в которое пользоватекль начал прослушаивать песню. 

Можно предположить, что идет сравнение пользователь в двух городах Спб и Мск, а также сравнение жанров по дням недели (какой жанр в какой день предпочтительнее).

Датасет содержит следующие колонки:
- userID - идентификатор пользователя,
- trackname - название песни,
- Artistname - исполнитель,
- genre - жанр,
- City - город,
- time - время,
- DayWeek - день недели.

In [10]:
#оценка данных
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65215 entries, 0 to 65214
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0     userID    65215 non-null  object
 1   trackname   63256 non-null  object
 2   Artistname  57330 non-null  object
 3   genre       63292 non-null  object
 4     City      64487 non-null  object
 5   time        64487 non-null  object
 6   Dayweek;;   64485 non-null  object
dtypes: object(7)
memory usage: 3.5+ MB


### С помощью метода .info оцените данные. 
### Этот метод возвращает название столбцов, типы данных, количество ненулевых объектов каждом столбце.

- В датасете содержится 65215 строк, 7 столбцов. 
- У всех полей тип данных - object. 
- Присувтствуют некорректные названия столбцов. 
- Количество значений в столбцах различается. Значит, в данных есть пропущенные значения.


In [12]:
#вывод названий столбцов
df.columns

Index(['  userID', 'trackname', 'Artistname', 'genre', '  City  ', 'time',
       'Dayweek;;'],
      dtype='object')

### Проблемы с названиями столбцов:
- присутсвие пробелов в названии(столбцы: city, userID),
- cтрочные буквы сочетаются с прописными(Artistname, City, Dayweek),
- лишние символы в названии столбца Dayweek,
- отсутсиве "змеиного стиля" (присутсвие нижнего подчеркивания) при названии из нескольких слов (userID, trackname, Artistname, Dayweek).


In [13]:
#приводим все наименования к нижнему регистру
df.columns = df.columns.str.lower()
df.columns

Index(['  userid', 'trackname', 'artistname', 'genre', '  city  ', 'time',
       'dayweek;;'],
      dtype='object')

In [14]:
#переименования столбцов
df=df.rename(columns={'  userid': 'user_id', 'trackname':
'track_name', 'artistname': 'artist_name', '  city  ': 'city', 'dayweek;;': 'day_week'})
df.columns


Index(['user_id', 'track_name', 'artist_name', 'genre', 'city', 'time',
       'day_week'],
      dtype='object')

In [11]:
df['trackname'] = df['trackname'].astype('string')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65215 entries, 0 to 65214
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0     userID    65215 non-null  object
 1   trackname   63256 non-null  string
 2   Artistname  57330 non-null  object
 3   genre       63292 non-null  object
 4     City      64487 non-null  object
 5   time        64487 non-null  object
 6   Dayweek;;   64485 non-null  object
dtypes: object(6), string(1)
memory usage: 3.5+ MB


## Поиск дубликатов данных